src/sentry/eventstream/kafka/protocol.py

```diff
@@ -172,9 +190,27 @@ def get_task_kwargs_for_message_from_headers(
                 "is_regression": is_regression,
                 "is_new_group_environment": is_new_group_environment,
             }
+
+            if "group_states" not in header_data:
+                header_data["group_states"] = None
+
+            group_states_str = None
```

Comment:
Nit: We don't need this here, but we could have this as `group_states = None` and remove that same line from the except blocks.